### ISSUE TITLE
[BUG] Equal Access Accessibility Checker errors when switching tabs and scanning complex/lengthy documents

### ISSUE THREAD

**Issue by thedigitalman** [IBMa/equal-access#497]
## What tool(s) are you using?

- [x] `accessibility-checker-extension`

## Detailed description

When **switching to the "Accessibility Assessment" tab in browser Dev Tools** I get an error in the Console. It happens in Chrome and Firefox.

Chrome (93):

```
TypeError: Cannot read property 'report' of undefined
    at tabListeners.js:1
    at Generator.next (<anonymous>)
    at tabListeners.js:1
    at new Promise (<anonymous>)
    at i (tabListeners.js:1)
    at tabListeners.js:1
    at tabListeners.js:1
(anonymous) @ tabListeners.js:1
(anonymous) @ tabListeners.js:1
i @ tabListeners.js:1
(anonymous) @ tabListeners.js:1
(anonymous) @ tabListeners.js:1
```

Firefox (92):

```
TypeError: t is undefined
    139 moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    i moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    i moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    139 moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    addListener moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
tabListeners.js:1:1897
    139 moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    i moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    i moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    139 moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    addListener moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
```

When **scanning a complex/lengthy document** I get an error in the Console. It happens in Chrome and Firefox.

Chrome (93):

```
tabListeners.js:1 TypeError: Cannot read property 'role' of undefined
    at t.e.closeScope (<anonymous>:16:97792)
    at e.run (<anonymous>:16:165855)
    at e.check (<anonymous>:16:162625)
    at tabListeners.js:1
    at Generator.next (<anonymous>)
    at tabListeners.js:1
    at new Promise (<anonymous>)
    at i (tabListeners.js:1)
    at tabListeners.js:1
    at tabListeners.js:1
(anonymous) @ tabListeners.js:1
(anonymous) @ tabListeners.js:1
i @ tabListeners.js:1
(anonymous) @ tabListeners.js:1
(anonymous) @ tabListeners.js:1
```

Firefox (92):

```
TypeError: t[(t.length - 1)] is undefined
    closeScope sandbox eval code:16
    run sandbox eval code:16
    check sandbox eval code:16
    139 moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    i moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    i moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    139 moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    addListener moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
tabListeners.js:1:2436
    139 moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    i moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    i moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    139 moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
    addListener moz-extension://53c24f3c-7c06-49fe-bf8d-9ffbfa3a7426/tabListeners.js:1
```

### Switching to the "Accessibility Assessment" tab in browser Dev Tools

1. Open Chrome/Firefox
1. Update extension settings/options
   1. Select "Latest Deployment" option in "Select a rule set deployment date" select list
   1. Select "WCAG 2.1 (A, AA)" option in "Select accessibility guidelines" select list
   1. Press "Save" button
1. Navigate to page
1. Open browser Dev Tools
1. Switch to "Accessibility Assessment" tab
1. Error appears in Console

### Scanning a complex document

1. Open Chrome/Firefox
1. Update extension settings/options
   1. Select "Latest Deployment" option in "Select a rule set deployment date" select list
   1. Select "WCAG 2.1 (A, AA)" option in "Select accessibility guidelines" select list
   1. Press "Save" button
1. Navigate to page with complex/lengthy document
1. Open browser Dev Tools
1. Switch to "Accessibility Assessment" tab
1. Press "Scan" button
1. Error appears in Console

## Additional information

Operating system: Windows 10 Enterprise (Version 1909, OS build 18363.1734)

Other browser extensions installed:

- Chrome
  - Awesome screenshot
  - axe DevTools
  - SEE
  - Tota11y
  - uBlacklist
  - Web Developer
  - Web Vitals
  - Windows 10 Accounts
- Firefox
  - Adblock
  - Augury
  - Awesome screenshot
  - Bitwarden
  - BuiltWith
  - Copy Selection as Markdown
  - Tota11y
  - uBlacklist
  - Web Developer

**Comment by aliunwala** [comment]
Do you have a link to the document that you ran this with. To help with reproduction of this error in the future. @thedigitalman

**Comment by tombrunet** [comment]
@thedigitalman Do you have an example page that we can see this on? We have not seen this error.

**Comment by aliunwala** [comment]
Unable to reproduce
